.github/workflows/pull-request-action.yml

```diff
@@ -0,0 +1,42 @@
+name: Reusable PR testing for mac and linux
+
+on:
+  workflow_call:
+    inputs:
+      targets:
+        type: string
+        description: "Make and CTest targets. If not specified, everything is tested"
+        required: false
+
+jobs:
+  handle-syncwith:
+    name: Call Reusable SyncWith Handler
+    uses: NilFoundation/ci-cd/.github/workflows/reusable-handle-syncwith.yml@v1.1.0
+    with:
+      ci-cd-ref: 'v1.1.3'
+    secrets: inherit
+
+  matrix-test-linux:
+    name: Linux Reusable Crypto3 Testing
+    needs:
+      - handle-syncwith
+    uses: NilFoundation/ci-cd/.github/workflows/reusable-crypto3-testing-linux.yml@f9d91d2a62f5fb8d6767e4a458cae49da29aefe9
```

Comment:
Friendly reminder to replace references. Btw, you can use branch name here, so you won't need to replace SHA manually